test/functional/api/v1/architectures_controller_test.rb

```diff
@@ -75,4 +75,12 @@ def user_one_as_manager
     end
     assert_response :success
   end
+
+  test "403 response contains missing permissions" do
```

Comment:
`test/functional/api/base_controller_subclass_test.rb` seems like a better fit for this kind of functionality that works all across the application. There are even some permissions related tests in there, it doesn't only belong on architecture.